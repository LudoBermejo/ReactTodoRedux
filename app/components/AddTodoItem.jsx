import React from 'react';

const AddTodoItem = (props) => {
  let textInput = '';

  const onSubmit = (e) => {
    e.preventDefault();
    if (textInput.value) {
      props.onAddItem(textInput.value);
      textInput.value = '';
    }
  };

  return (
    <div className="container__footer" >
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="What do you need to do?"
          ref={(input) => { textInput = input; }}
        />
        <button className="button expanded">Add todo</button>
      </form>
    </div>
  );
};

AddTodoItem.propTypes = {
  onAddItem: React.PropTypes.func.isRequired
};

export default AddTodoItem;
