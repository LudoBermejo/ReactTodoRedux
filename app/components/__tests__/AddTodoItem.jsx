import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import AddTodoItem from 'AddTodoItem';

describe('AddTodoItem ', () => {
  it('renders properly', () => {

    const mock = jest.fn();
    const addTodoItem = renderer.create(<AddTodoItem onAddItem={mock} />).toJSON();
    expect(addTodoItem).toMatchSnapshot();
  });

  describe('works as expected', () => {
    it('doesn\'t call onAddItem if value is empty', () => {
      const mock = jest.fn();
      const addTodoItem = shallow(<AddTodoItem onAddItem={mock} />);
      addTodoItem.find('button').simulate('submit');
      expect(mock).not.toHaveBeenCalled();
    })

    it('call onAddItem if value is empty', () => {
      const mock = jest.fn();
      const addTodoItem = mount(<AddTodoItem onAddItem={mock} />);
      const first = addTodoItem.find('input');
      first.node.value = 'fff';
      first.simulate('change', first);
      addTodoItem.find('form').simulate('submit');
      expect(mock).toHaveBeenCalledWith('fff');
      expect(first.node.value).toBe('');
    })
  })
});

