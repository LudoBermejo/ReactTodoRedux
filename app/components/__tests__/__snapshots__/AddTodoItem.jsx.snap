exports[`AddTodoItem  renders properly 1`] = `
<form
  className="form"
  onSubmit={[Function]}>
  <input
    placeholder="What do you need to do?"
    type="text" />
  <button
    className="button expanded">
    Add todo
  </button>
</form>
`;
